### PR TITLE
implement flag to allow changing the default StorageClass after migration completion

### PR DESCRIPTION
**Pull request by laverya**

### CODE REVIEW

File: pkg/migrate/migrate.go [hunk 1]
```diff
@@ -485,6 +538,50 @@ func mutatePV(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
...
+	for {
+		sc, err := clientset.StorageV1().StorageClasses().Get(ctx, scName, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get storage classes %s: %w", scName, err)
+		}
+
+		if checker(sc) {
+			return nil
+		}
+		time.Sleep(time.Second * 5)
```
Comment (emosbaugh) [inline pkg/migrate/migrate.go:581]: is this supposed to try forever?

File: pkg/migrate/migrate.go [hunk 2]
```diff
@@ -104,10 +108,63 @@ func Migrate(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
...
 	return nil
 }
 
+func swapDefaults(ctx context.Context, w *log.Logger, clientset k8sclient.Interface, oldDefaultSC string, newDefaultSC string) error {
+	// create a pod for each PVC migration, and wait for it to finish
+	w.Printf("\nChanging default StorageClass from %s to %s\n", oldDefaultSC, newDefaultSC)
+	err := mutateSC(ctx, w, clientset, oldDefaultSC, func(sc *storagev1.StorageClass) (*storagev1.StorageClass, error) {
+		// check to make sure that this is actually the default StorageClass before deleting the annotation
+		if sc.Annotations == nil {
+			return nil, fmt.Errorf("%s is not the default StorageClass", oldDefaultSC)
```
Comment (emosbaugh) [inline pkg/migrate/migrate.go:128]: what if this was the default but the script failed the first time before setting the new default? or is that not a case we want to handle?

Comment (laverya) [reply pkg/migrate/migrate.go:128]: There shouldn't be a way to fail after successfully applying by my reading (anything that returns an error shouldn't update the state of the object)

File: pkg/migrate/migrate.go (hunk 1 above)
Comment (laverya) [reply pkg/migrate/migrate.go:581]: If this takes more than a minute, something has gone badly wrong - but I don't know if that's worth handling

File: pkg/migrate/migrate.go (hunk 2 above)
Comment (emosbaugh) [reply pkg/migrate/migrate.go:128]: you remove the annotation from the storage class first then add to the new storage class. if you fail to add but successfully remove wont the script fail the next time with is not the default StorageClass?

its an edge case. but i thought worth noting. im not gonna block the pr for it but if you choose not to handle it now maybe add a comment?

File: pkg/migrate/migrate.go (hunk 1 above)
Comment (emosbaugh) [reply pkg/migrate/migrate.go:581]: makes sense

File: pkg/migrate/migrate.go (hunk 2 above)
Comment (laverya) [reply pkg/migrate/migrate.go]: Added a bunch of code to handle this (no existing default StorageClass) and also the case where the desired dest StorageClass was already the default

**Review by emosbaugh — APPROVED** [review]

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**